src/robotide/ui/tagdialogs.py

```diff
@@ -100,6 +101,14 @@ def _make_bindings(self):
         self._tags_list.Bind(wx.EVT_LIST_ITEM_RIGHT_CLICK, self.OnRightClick)
         self._tags_list.Bind(wx.EVT_LIST_COL_CLICK, self.OnColClick)
 
+    def _tag_name_for_sort(self, tag_name):
+        # Make tag_name lowercase for the sorting algorithm only
+        tag_name = tag_name.lower()
+        # Compare numbers as numeric values
+        m = re.search(r'([^0-9]*)([0-9]*)', tag_name)
+        g = m.groups()
+        return g[0], int(g[1] or 0)
```

Comment:
The following images show the differences:
Before:
![tag_sort_old_algorithm](https://cloud.githubusercontent.com/assets/2866630/13459318/9cf66eee-e06a-11e5-81c1-efac42bb2caf.png)
After:
![tag_sort_new_algorithm](https://cloud.githubusercontent.com/assets/2866630/13459329/ab011ea8-e06a-11e5-82ff-cf934a72a802.png)

Is it acceptable?